comment/api/serializers.py

```diff
@@ -118,6 +118,8 @@ def create(self, validated_data):
         user = get_user_for_request(request)
         content = validated_data.get('content')
         email = validated_data.get('email')
+        if not user:
+            self.validate_email(email)
```

Comment:
i don't think this is required. the `email` field will only be added in the `serializer` when user is unauthenticated and commenting by anonymous is allowed:  https://github.com/Radi85/Comment/pull/201/files#diff-580565eef493475112cb9daed5cc31fd758b926848bc1ba626bab78ea81acc96R102-R106

in the scenario when `email` field is added to the `serializer` fields, the `validate_email` method is already called by the parent `serializer.Serializer` class. See the documentation regarding this here: https://www.django-rest-framework.org/api-guide/serializers/#field-level-validation

Reply:
Hmmm as per REST [documentation](https://www.django-rest-framework.org/api-guide/serializers/#field-level-validation) this method is only triggered when the field is required. So in our case we just need to set the field to required when anonymous is used.

```python
        if not user.is_authenticated and settings.COMMENT_ALLOW_ANONYMOUS:
            self.fields['email'].required = True
```
Will do the trick.

Reply:
i think by default it is required is set to `True` or whatever the model field says. also, as i see there is an `DABEmailValidator` already used in this view, where empty values which won't allow empty values. 
For better error message maybe we can move the code from `validate_email` function there?

Reply:
It's `False` by default, setting it to `True` will not hit the `validate_email` either but the email however validated with the default message. I am going to keep calling `validate_email` inside `create` method and I see that we no longer need the `init` method.